include/aspect/mesh_deformation/interface.h

```diff
@@ -651,8 +651,8 @@ namespace aspect
           if (Plugins::plugin_type_matches<MeshDeformationType>(*p))
             return Plugins::get_plugin_as_type<MeshDeformationType>(*p);
 
-      typename std::vector<std::unique_ptr<Interface<dim>>>::const_iterator mesh_def;
       // We will never get here, because we had the Assert above. Just to avoid warnings.
+      typename std::vector<std::unique_ptr<Interface<dim>>>::const_iterator mesh_def;
       return Plugins::get_plugin_as_type<MeshDeformationType>(*(*mesh_def));
```

Comment:
Why not use the same idiom here?

Reply:
because this one is different: there are to nested objects and it would be something like:
***mesh_deformation_objects.begin().second